cst438Assignment2/src/main/java/cst438hw2/service/CityService.java

```diff
@@ -0,0 +1,44 @@
+package cst438hw2.service;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import cst438hw2.domain.*;
+
+@Service
+public class CityService
+{
+   @Autowired
+   private CityRepository cityRepository;
+   
+   @Autowired
+   private CountryRepository countryRepository;
+   
+   @Autowired
+   private WeatherService weatherService;
+   
+   
+   public CityInfo getCityInfo(String cityName) {
+     List<City> cities = cityRepository.findByName(cityName);
+     
+     if(cities.size() == 0) {
+        return null;
+     }
+     
+     City city = cities.get(0);
+     Country country = city.getCountry();
+   //Country country = countryRepository.findByCode(city.getCountry())
+     //Do we even need countryRepository???
```

Comment:
One way I tried to use CountryRepository was once you get values from city.getCountry, you can then call .getCode() which would return a country code because city.getCountry returns an Object like Country [ "AUS", "Australia"] so by nesting it like city.getCountry().getCode() it then returns just "AUS" which would allow you to use countryRepository.findByCode